### ISSUE TITLE
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /www/orbitdb/node_modules/ipfs-http-client/package.json

### ISSUE THREAD

**Issue by rrubio** [ipfs/js-ipfs#4139]
IPFS Version - 

- "ipfs-http-client": "^57.0.1",

NodeJS Version - 

- v14.19.3

NPM Version - 

- 6.14.17

OS -

- macOS Monterey Version 12.4

Error occurs when i run my script like so "node client.js" - 

`
const IpfsClient = require('ipfs-http-client')
const OrbitDB = require('orbit-db')

const ipfs = IpfsClient('localhost', '5001')

(async () => {

    const orbitdb = await OrbitDB.createInstance(ipfs)

    const db = await orbitdb.log('hello-world')

})()
`

I have tried adding the following to "node_modules -> ipfs-http-client -> package.json" with no luck - 

`  "exports": {
    ".": {
      "import": "./src/index.js"
    },
    "./client": "./client.js"
`

Please note - i get the exact same error when using "ipfs".

**Comment by achingbrain** [comment]
The latest `ipfs-http-client` is ESM only.  This means you cannot use `require` to load the module, you must use `import`.

Please see the [upgrade guide](https://github.com/ipfs/js-ipfs/blob/master/docs/upgrading/v0.62-v0.63.md#esm).

**Comment by ffarall** [comment]
Hey! I am using `import { create } from "ipfs-http-client";` in my project, and I still get this error, any idea why that could be?

**Comment by nanoTitan** [comment]
I'm seeing the same thing. I using es6 import and getting that same error:
import { create } from 'ipfs-http-client';

No "exports" main defined in ...\node_modules\ipfs-http-client\package.json

**Comment by achingbrain** [comment]
If you're doing something like using typescript with `import` in your `.ts` files and still seeing `No "exports" main defined in`, chances are your tsc config is set to output CJS.  It needs to be ESM - set this using the ["module"](https://www.typescriptlang.org/tsconfig#module) key in `tsconfig.json` - it should be `"es2020"` at least.

**Comment by nanoTitan** [comment]
Yes, that is the problem I figured out just recently. I'm using NestJS which doesn't support ESM modules right now. Also, doing dynamic imports with `const { create } = await import('ipfs-http-client')` isn't working for me either.

**Comment by KKRAKOVIA** [comment]
> Hey! I am using `import { create } from "ipfs-http-client";` in my project, and I still get this error, any idea why that could be?
> 
> It seems to be due to the fact that I'm using it in a [HardHat](https://hardhat.org/) task, and that is not an ESM, even though I am using `import` for everything.

had same problem with HH right now, but i used that module in the past.
Resolved with
pnpm i ipfs-http-client@56.0.2

**Comment by SiIith** [comment]
> pnpm i ipfs-http-client@56.0.2

This solved our problem when using nestJs, thanks!

**Comment by kc97ble** [comment]
> > Hey! I am using `import { create } from "ipfs-http-client";` in my project, and I still get this error, any idea why that could be?
> > It seems to be due to the fact that I'm using it in a [HardHat](https://hardhat.org/) task, and that is not an ESM, even though I am using `import` for everything.
> 
> had same problem with HH right now, but i used that module in the past. Resolved with pnpm i ipfs-http-client@56.0.2

Downgrading to version 56.x.x works for me.

**Comment by rahulrumalla** [comment]
Same here. Downgrading to ipfs-http-client@56.0.2 worked for me. Will there be a fix for this in the higher versions?

**Comment by bhflm** [comment]
Confirm, was having this problem until I did downgrade to 56.0.2 and it worked :) 👍🏼

**Comment by josh-richardson** [comment]
I'm also hitting this when using ipfs-http-client in Nest, with typescript and only importing using `import { create } from 'ipfs-http-client';`.

**Comment by hubsmoke** [comment]
Please see this comment https://github.com/ipfs/js-ipfs/issues/4180#issuecomment-1319951128

**Comment by BigBen3918** [comment]
@KKRAKOVIA Thank you.
Issue solved. Great!

**Comment by francesco-gaglione** [comment]
Downgrading to v56.0.2 and with this code:

```
let ipfs: IPFSHTTPClient = create({
                host: "ipfs.infura.io",
                port: 5001,
                protocol: "https",
                headers: {
                    authorization: this.auth
                }
            });


            ipfs.add("test").then((res)=>{
                resolve(res.cid.toString());
            })
```

I have this error:

`TypeError: RequestInit: duplex option is required when sending a body.`

It seem to be a node fetch error, how to solve?

**Comment by Raman-Raje** [comment]
@francesco-gaglione I am also getting the same error...
Were you able to resolve it....?

**Comment by rafinskipg** [comment]
I'm getting the same error

**Comment by mime29** [comment]
Same here and the only way I found to fix it was to modify the source ->
<img width="300" alt="image" src="https://user-images.githubusercontent.com/2211828/216766857-21c97d9a-75cf-4355-9a38-8841bb17c1c0.png">

**Comment by phungson185** [comment]
> > Hey! I am using `import { create } from "ipfs-http-client";` in my project, and I still get this error, any idea why that could be?
> > It seems to be due to the fact that I'm using it in a [HardHat](https://hardhat.org/) task, and that is not an ESM, even though I am using `import` for everything.
> 
> had same problem with HH right now, but i used that module in the past. Resolved with pnpm i ipfs-http-client@56.0.2

Resolved my problem
